### DISCUSSION TITLE
Link with padding overflows container

### DISCUSSION THREAD

**Discussion by spiffytech** [tailwindlabs/tailwindcss#10083] [Help] [answered]
[Sandbox](https://codepen.io/spiffytech/pen/JjZgKEY)

If I add y-padding to an `<a>` the padding overflows the links' container and covers up adjacent elements.

This also affects placement; the link's top is calculated from the _original_, unpadded top.

If I remove the Tailwind styles the link is positioned correctly.

I can mark the link as `inline-block`, but I'm not sure why that's necessary, and in my real use case I want an inline element anyway.

**Comment by ProPhycient5** [comment]
Hi @spiffytech, Can you explain your doubt in a simpler form by uploading any screenshots of what you want to achieve and the current case?

**Reply by spiffytech** [reply]
I've now uploaded a screenshot below.

**Comment by wongjn** [comment, marked as answer]
The behavior you are experiencing is precisely because `<a>` elements are inline elements and this is how inline elements behave with respect to vertical padding, since inline elements will try to stay on the text baseline as much as possible.

As you say, you could apply `display: inline-block` via `inline-block` to have the `<a>` element's layout box not cover the previous element's layout box.

**Reply by spiffytech** [reply]
Here's how the browser styles this by default:
![image](https://user-images.githubusercontent.com/77867/207874711-2cb5b71b-e5d6-4ae6-9f81-f3fa375f0290.png)

Here's what the Tailwind CSS reset does (no other changes):
![image](https://user-images.githubusercontent.com/77867/207874620-729bea7f-100a-4387-a3bd-31d5fe21d0b4.png)

Why does the Tailwind CSS reset change the behavior?

**Reply by wongjn** [reply]
With no Tailwind styling, the `<p>` element typically has bottom margin from the user-agent stylesheet which pushes the `<a>` element down. With Tailwind's preflight styling, it removes margins declared from the user-agent stylesheet from most (if not all) elements. I cannot authoritatively say why, but for me this is a desired reset such that I can more reliably space all block elements "from 0".

If you would not like this behavior, [you can disable the core preflight plugin](https://tailwindcss.com/docs/preflight#disabling-preflight).

**Reply by spiffytech** [reply]
Ahhhh that makes so much sense! Thanks for solving the mystery :smile:
